### ISSUE TITLE
Use a faster json lib?

### ISSUE THREAD

**Issue by NamedKitten** [bwmarrin/discordgo#463]
I was wondering if it would be worth using this.
https://github.com/json-iterator/go

**Comment by SilverCory** [comment]
Make a PR?
I'd like to see this implemented benchmarked and compared personally. But I don't see the library as being slow at all.

**Comment by bwmarrin** [comment]
Yeah personally I'd prefer to avoid any more dependencies on other 3rd party libraries.  So unless there's an overwhelming need/reason and the 3rd party library looks like something that will be around for awhile, I'd prefer to just use the built-in json functions.  

So, if you want to make a PR with benchmarks of both options and also discuss if that performance has a real world need - that would be the next step here.

**Comment by jonas747** [comment]
I tried it a while back with discordgo, it was around 3x faster decoding incoming events, it wasn't entirely drop-in as advertised and i never finished completing the transition completely, if that's enough of a benefit i can complete that transition and pr it.

**Comment by toby3d** [comment]
How about this library: https://github.com/pquerna/ffjson
I use this with [fasthttp](/valyala/fasthttp) in my projects and get x2-3 more speed.

**Comment by tsdtsdtsd** [comment]
Last time I tried a few json libs my conclusion was that performance benefits are minimal for shallow json documents with depth <= 2, maybe 3.
You only get a nice boost with heavily nested documents. Tbh I never tested "large" documents with lots of fields. Benchmarks to the rescue! That said, I have no clue how the json we're dealing with looks like... Maybe I'll find the time to log a few real docs and write some benchmarks.
